dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/user/PasswordValidationRuleTest.java

```diff
@@ -101,6 +102,23 @@ public void init()
         historyValidationRule = new PasswordHistoryValidationRule( passwordEncoder, userService, currentUserService );
     }
 
+    @Test
+    public void testWhenPasswordIsNullOrEmpty()
```

Comment:
You are testing also an empty email. So I would use a better name for the test method name. It says nothing about email. Eventually, create a separate test case.